### ISSUE TITLE
Support images transformation

### ISSUE THREAD

**Issue by lane711** [lane711/sonicjs#243]
Now that we support image management, it makes sense to extend this to supporting image transformation. 

So for example, a user uploads a full rez photo, and SonicJs will create various versions of the images based on preset configurations including:
1. image resolution
2. width
3. height

The configuration should  be stored as env vars at this stage

For reference, please see how this is handled in storyblok:
https://www.storyblok.com/tp/optimize-your-storyblok-images-with-image-engine

**Comment by jamaluddinrumi** [comment]
✨amazing 

more storyblok users & similar headless cms users will come to here 😁

**Comment by jamaluddinrumi** [comment]
sorry, i put the wrong reference url. 

the right one is here https://www.storyblok.com/docs/image-service

it contains more comprehensive & demos on how storyblok generate dynamic responsive images based on url params.

**Comment by osseonews** [comment]
Instead of using R2 to store images, why not just use Cloudflare Images, https://developers.cloudflare.com/images/. This will automatically transform the images for you. I don't know why anyone uses R2 for images actually. It's better for documents. So in your repo, when someone uploads an image from TUS just use the cloudflare images API to store there, and then you get automatic transformations out of the box. Easy to use.

**Comment by lane711** [comment]
I didn't even know about cf images. That is something we should be using for sure. I'll add an issue for that. Thanks for letting us know

**Comment by osseonews** [comment]
Yeah, basically Cloudflare Images is built on top of R2 and just includes all the other features that everyone needs for images that they would have to build on their own with just R2. So you don't have to reinvent the wheel with R2 for tranformations etc., as Cloudflare has already built it. It's an amazing service. I think you can keep R2 here, but just use it for files, like PDF uploads. Images should just use Cloudflare Images as it has every thing you can possibly want for images (and backend is R2).

**Comment by lane711** [comment]
Closing this issue as it's no longer relevant following the SonicJS 2.0 rewrite. The 2.0 version is a complete architectural overhaul with significant improvements to the core framework. If you encounter similar issues with 2.0, please open a new issue with updated details. Thank you!
